### PR TITLE
Unlocked Stage Activity

### PR DESCRIPTION
**Pull request by Raxerz**
This PR contains files for the last stage activity.
Here is a video of the same: https://drive.google.com/file/d/0B7Hnx8cG8DNBMmludy1qMl9WMnc/view?usp=sharing

### CODE REVIEW

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        <div id="human-health">
+            <div id="human-health-bar"></div>
+        </div>
+        <div id="mosquito-health-text">{{language?.legend.mosquito}}</div>
+        <div id="mosquito-health">
+            <div id="mosquito-health-bar"></div>
+        </div>    
+        <img id="human" src="../assets/img/unlocked-stage/human.png"/>
+        <img id="moz" name="animate" src="../assets/img/unlocked-stage/mosquito.gif"/>  
+        <img id="spray" src="../assets/img/unlocked-stage/spray-cursor.gif" style="display:none"/>
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: inline style

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    </div>
+    <div id="inventory">
+        <div id="inventory-heading">{{language?.legend.inventory}}</div>
+        <img id="spray-item" src="../assets/img/unlocked-stage/spray-item.gif"/>
+    </div>
+</div>
+<div class="row col-lg-6 col-lg-offset-3 text-center" id="game-end-status" [hidden]="!gameOver">
+    <div class="swal2-icon swal2-error swal2-animate-error-icon" style="display: block;">
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: inline style

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        <span class="swal2-x-mark swal2-animate-x-mark">
+            <span class="swal2-x-mark-line-left"></span>
+            <span class="swal2-x-mark-line-right"></span>
+        </span>
+    </div>
+    <div>
+        <h2 class="swal2-title text-center" id="swal2-title">{{language?.alerts.loss.title}}</h2>
+    </div>
+    <div>
+        <h2 class="swal2-title text-center" id="swal2-title">{{language?.alerts.loss.content}}</h2>
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: duplicated id

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        <img id="trans-gif" src="../assets/img/unlocked-stage/mosquito-transmission.gif"/>
+    </div>
+    <div class="swal2-title text-center" id="restart">
+        <button class="btn btn-primary btn-lg" (click)="reload()">{{language?.messages.restartButton}}</button>
+    </div>
+</div>
+
+<div class="row col-lg-6 col-lg-offset-3 text-center" id="game-end-status" [hidden]="!success">
+    <div class="swal2-icon swal2-success swal2-animate-success-icon" style="display: block;">
+        <div class="swal2-success-circular-line-left" style="background: rgb(255, 255, 255);"></div>
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: inline style

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        <span class="swal2-success-line-tip swal2-animate-success-line-tip"></span> 
+        <span class="swal2-success-line-long swal2-animate-success-line-long"></span>
+        <div class="swal2-success-ring"></div> 
+        <div class="swal2-success-fix" style="background: rgb(255, 255, 255);"></div>
+        <div class="swal2-success-circular-line-right" style="background: rgb(255, 255, 255);"></div>
+    </div>
+    <div>
+        <h2 class="swal2-title text-center" id="swal2-title">{{language?.alerts.win.title}}</h2>
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: id

File: src/app/unlocked-stage/unlocked-stage.component.scss [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    }
+}
+
+
+@mixin health($top, $left) {
+    @include positioning(absolute, $top, $left);
+    width: 20%;
+    background-color: white;
+    @include borderRadius(25px);
+    border: 1px solid #000000;
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.scss]: `black`?

File: src/app/unlocked-stage/unlocked-stage.component.scss [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+#restart {
+    padding-top: 5%;   
+}
+
+#inventory {
+    @include positioning(absolute, 125px, 83%);
+    border: 1px solid black;
+    width: 8%;
+    background: #FAE3BB;
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.scss]: variable?

**Comment by Raxerz** [comment]
Can you please check now @yukie1412

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        <div class="swal2-success-circular-line-right"></div>
+    </div>
+    <div>
+        <h2 class="swal2-title text-center">{{language?.alerts.win.title}}</h2>
+    </div>
+    <div>
+        <h2 class="swal2-title text-center">{{language?.alerts.win.content}}</h2>
+        <img id="prevention" src="../assets/img/unlocked-stage/malaria-prevention.png"/>
+    </div>
+    <div class="swal2-title text-center" id="restart">
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: duplicated id

File: src/app/unlocked-stage/unlocked-stage.component.scss [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    &:before {
+        @include beforeElement(8px)
+    }
+}
+
+
+@mixin health($top, $left) {
+    @include positioning(absolute, $top, $left);
+    width: 20%;
+    background-color: white;
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.scss]: `$white?`

File: src/app/unlocked-stage/unlocked-stage.component.scss [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+@mixin legendText($top, $left) {
+    @include positioning(absolute, $top, $left);
+    font-size: 1vw;
+    font-weight: 900;
+}
+
+#scene {
+    border: 1px solid black;
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.scss]: `$black`?

**Comment by Raxerz** [comment]
Made the changes @yukie1412 . Please check

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    <div>
+        <h2 class="swal2-title text-center">{{language?.alerts.loss.content}}</h2>
+        <img id="trans-gif" src="../assets/img/unlocked-stage/mosquito-transmission.gif"/>
+    </div>
+    <div class="swal2-title text-center restart">
+        <button class="btn btn-primary btn-lg" (click)="reload()">{{language?.messages.restartButton}}</button>
+    </div>
+</div>
+
+<div class="row col-lg-6 col-lg-offset-3 text-center" id="game-end-status" [hidden]="!success">
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: this id still is the same as the div above

File: src/app/unlocked-stage/unlocked-stage.component.html [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+</div>
+<div class="row col-lg-6 col-lg-offset-3 text-center" id="game-end-status" [hidden]="!gameOver">
+    <div class="swal2-icon swal2-error swal2-animate-error-icon">
+        <span class="swal2-x-mark swal2-animate-x-mark">
+            <span class="swal2-x-mark-line-left"></span>
+            <span class="swal2-x-mark-line-right"></span>
+        </span>
+    </div>
+    <div>
+        <h2 class="swal2-title text-center">{{language?.alerts.loss.title}}</h2>
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.html]: out of curiosity, is the text centered if you remove `.text-center`?

File: src/app/unlocked-stage/unlocked-stage.component.ts [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+            if (currObj._mosquitoBarWidth <= 0) {
+                currObj.success = true;
+                currObj._dashboardService.updateProgressStatus(currObj._status).subscribe(response => {});
+                return;
+            }
+            if (currObj._humanBarWidth <= 0) {
+                currObj.gameOver = true;
+                return;
+            }
+            if (currObj._spray && currObj._mousedown) {
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.ts]: `if ((currObj._spray && currObj._mousedown) && (currObj.checkCollision(currObj._spray, currObj.mosquito)) {`?

File: src/app/unlocked-stage/unlocked-stage.component.ts [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+        const leftObj1 = obj1.offsetLeft;
+        const bottomObj1 = topObj1 + obj1.height;
+        const rightObj1 = leftObj1 + obj1.width;
+        const topObj2 = obj2.offsetTop;
+        const leftObj2 = obj2.offsetLeft;
+        const bottomObj2 = topObj2 + obj2.height;
+        const rightObj2 = leftObj2 + obj2.width;
+        if (!(bottomObj1 < topObj2 || topObj1 > bottomObj2 || leftObj1 > rightObj2 || rightObj1 < leftObj2)) {
+            return true;
+        }
```
Comment (yukie1412) [inline src/app/unlocked-stage/unlocked-stage.component.ts]: should have a `return false;` as default.

File: src/app/unlocked-stage/unlocked-stage.component.html (hunk 12 above)
Comment (Raxerz) [inline src/app/unlocked-stage/unlocked-stage.component.html]: No.
The text is not centered if .text-center is removed.
Its a bootstrap class actually.

File: src/app/unlocked-stage/unlocked-stage.component.ts (hunk 14 above)
Comment (Raxerz) [inline src/app/unlocked-stage/unlocked-stage.component.ts]: Made the change.

**Comment by Raxerz** [comment]
Resolved @yukie1412

**Pull request merged**